dotnet/src/Connectors/Connectors.Memory.Qdrant/QdrantMemoryStore.cs

```diff
@@ -0,0 +1,369 @@
+        }
+        catch (Exception ex)
+        {
+            throw new VectorDbException($"Failed to remove vector data from Qdrant {ex.Message}");
+        }
+    }
+
+    /// <inheritdoc />
+    public async Task RemoveBatchAsync(string collectionName, IEnumerable<string> keys, CancellationToken cancel = default)
+    {
+        foreach (var key in keys)
+        {
+            await this.RemoveAsync(collectionName, key, cancel);
+        }
+    }
+
+    /// <summary>
+    /// Remove a MemoryRecord from the Qdrant Vector database by pointId.
+    /// </summary>
+    /// <param name="collectionName"></param>
+    /// <param name="pointId"></param>
+    /// <param name="cancel"></param>
+    /// <returns></returns>
+    /// <exception cref="VectorDbException"></exception>
+    public async Task RemoveWithPointIdAsync(string collectionName, string pointId, CancellationToken cancel = default)
+    {
+        try
+        {
+            await this._qdrantClient.DeleteVectorsByIdAsync(collectionName, new[] { pointId }, cancel: cancel);
+        }
+        catch (Exception ex)
+        {
+            throw new VectorDbException($"Failed to remove vector data from Qdrant {ex.Message}");
+        }
+    }
+
+    /// <summary>
+    /// Remove a MemoryRecord from the Qdrant Vector database by given a group of pointIds.
+    /// </summary>
+    /// <param name="collectionName"></param>
+    /// <param name="pointIds"></param>
+    /// <param name="cancel"></param>
+    /// <returns></returns>
+    /// <exception cref="VectorDbException"></exception>
+    public async Task RemoveWithPointIdBatchAsync(string collectionName, IEnumerable<string> pointIds, CancellationToken cancel = default)
+    {
+        try
+        {
+            await this._qdrantClient.DeleteVectorsByIdAsync(collectionName, pointIds, cancel: cancel);
+        }
+        catch (Exception ex)
+        {
+            throw new VectorDbException($"Error in batch removing data from Qdrant {ex.Message}");
+        }
+    }
+
+    public async IAsyncEnumerable<(MemoryRecord, double)> GetNearestMatchesAsync(
+        string collectionName,
+        Embedding<float> embedding,
+        int limit,
+        double minRelevanceScore = 0,
+        [EnumeratorCancellation] CancellationToken cancel = default)
+    {
+        var results = this._qdrantClient.FindNearestInCollectionAsync(
+            collectionName: collectionName,
+            target: embedding.Vector,
+            threshold: minRelevanceScore,
+            top: limit,
+            cancel: cancel);
+
+        await foreach ((QdrantVectorRecord, double) result in results)
+        {
+            yield return (
+                MemoryRecord.FromJson(
+                    json: result.Item1.GetSerializedPayload(),
+                    embedding: new Embedding<float>(result.Item1.Embedding)),
+                result.Item2);
+        }
+    }
+
+    public async Task<(MemoryRecord, double)?> GetNearestMatchAsync(
```

Comment:
It would be nice to XMLDoc for all public methods in this class that are not inheriting them from the interface. It can really help those new to the codebase and concepts to onboard.

Reply:
oop, I think I missed an <inheritdoc/> here